src/Runner/ResultCache/ResultCache.php

```diff
@@ -18,13 +18,13 @@
  */
 interface ResultCache
 {
-    public function setStatus(string $id, TestStatus $status): void;
```

Comment:
this might be a bit controveral, but IMO using a dedicated `ResultCacheId` class should help against passing wrongly formatted strings into the `ResultCache` in the future